pages/blog/swr-v2.en-US.mdx

```diff
@@ -19,13 +19,155 @@ import Video from 'components/video'
+
+When clicking the button, `sendRequest` will be triggered with the extra argument `{ username: 'johndoe' }`, and `isMutating` will be set to `true` until the mutation is finished.
+
+This new hook also covers other pain points you might have for mutations:
+
+- Show optimistic UI while mutating
+- Rollback automatically when mutation fails
+- Detect and avoid race conditions between `useSWR` and other mutations of the same resource
+- Populate the cache of `useSWR` after mutation finishes
+- ...
+
+Detailed API references and examples can be found [here](/docs/mutation#useswrmutation) and following sections.
+
+### Optimistic UI for Mutations
+
+Optimistic UI is a great model to make your website feel fast and snappy, but at the same time very hard to do correct. 
+SWR has some new powerful options added to make it easier.
+
+For example, we have a normal mutation that adds a new todo to the todo list. The `await addNewTodo(...)` request
+could be slow sometimes, so we want to immediately show the new item everywhere in the UI. The new `optimisticData` option does exactly that:
+it updates the local SWR cache optimistically.
+
+Then, let’s say when the request finishes, `await addNewTodo(...)` request directly returns the updated list. In that case we can
+enable `populateCache` to update our data directly using that response. As it’s already from the source of the truth,
+we don’t need another revalidation afterwards, hence we disabled `revalidate`.
+
+Lastly, if `await addNewTodo(...)` fails with an exception, we can rollback with `rollbackOnError` set to `true` (which is also the default).
+The `optimisticData` we previously displayed will be reverted.
+
+```jsx
+const { mutate, data } = useSWR('/api/todos')
+
+// ...
+mutate(async () => await addNewTodo('New Item'), {
+  optimisticData: [...data, 'New Item'],
+  populateCache: true,
+  revalidate: false,
+  rollbackOnError: true,
+})
+```
+
+And here is a demo showing that behavior:
+
+<Video
+  src="/video/optimistic-ui.mp4"
+  caption="Optimistic UI with automatic error rollback"
+  ratio={2348/896}
+/>
+
+These APIs are supported in the new `useSWRMutation` hook as well. To learn more about them, you can check out our [docs](/docs/mutation#optimistic-updates).
+
+### Mutate Multiple Keys
+
+The global `mutate` API now accepts a filter function, where you can mutate or revalidate specific keys. This will be helpful for use cases such as invalidting all the cached data. To learn more, you can read [Mutate Multiple Keys](/docs/mutation#mutate-multiple-items) in the docs.
+
+```jsx
+import { mutate } from 'swr'
+// Or from the hook if you have customized your cache provider:
+// { mutate } = useSWRConfig()
+
+// Mutate single resource
+mutate(key)
+
+// Mutate multiple resources to make them `undefined`
+mutate(
+  key => typeof key === 'string' && key.startsWith('/api/item?id='),
+  undefined,
+  { revalidate: true }
+)
+```
+
+## Better User Experience
```

Comment:
IMO, "SWR DevTools" and "Extending SWR Configurations" improve Developer Experience rather than User Experience. Does "User" mean developers?

Reply:
👍 
```suggestion
## Better Developer Experience
```

Reply:
"Prealod", "Loading State", "Laggy UI" improve user experience, so it might be better to split the section like this.

- Better User Experience
  -  Preload Data
  - New Loading State Indicator
  - “Laggy UI”
- Better Developer Experience
  -  SWR DevTools
  - Extending SWR Configurations